static/js/stream_settings_data.js

```diff
@@ -6,14 +6,10 @@ import * as stream_data from "./stream_data";
 import * as util from "./util";
 
 export function get_sub_for_settings(sub) {
-    // Since we make a copy of the sub here, it may eventually
-    // make sense to get the other calculated fields here as
-    // well, instead of using update_calculated_fields everywhere.
-    const sub_count = peer_data.get_subscriber_count(sub.stream_id);
-    return {
-        ...sub,
-        subscriber_count: sub_count,
-    };
+    const settings_sub = {...sub};
+    add_settings_fields(settings_sub);
+    settings_sub.subscriber_count = peer_data.get_subscriber_count(sub.stream_id);
+    return settings_sub;
 }
```

Comment:
This last commit may benefit from a commit message mention that doing these calculations at time of use will likely decrease the set of bugs we'd need to resolve if we wanted to support life-update when the current user's role changes correctly, since we're no longer caching these computed fields on `sub` objects.